### PR TITLE
Remove closing slash as per Joomla Coding Guidelines

### PR DESCRIPTION
**Pull request by PhilETaylor**
Just removing a closing slash from an img tag to make @C-Lodder a happier little boy

Ensure user provided data is correctly encoded.

### CODE REVIEW

File: plugins/fields/media/tmpl/media.php [hunk 1]
```diff
@@ -31,7 +30,9 @@
 		continue;
 	}
 
-	$buffer .= '<img src="' . htmlentities($path) . '"' . $class . '>';
+	$buffer .= sprintf('<img src="%s"%s>',
+			htmlentities($path, ENT_COMPAT, 'UTF-8', true),
+			$class );
```
Comment (Quy) [inline plugins/fields/media/tmpl/media.php]: Dedent lines 34 and 35.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on a1e0b51adfbdccf01843741baa42f8086ac6e74d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16151">issues.joomla.org/tracker/joomla-cms/16151</a>.</sub>

**Comment by Quy** [comment]
In `\plugins\fields\imagelist\tmpl\imagelist.php`, `directory` is accounted for. Does it apply here? If yes, probably should be in a separate PR.

```
	if (!$path || $path == '-1')
	{
		continue;
	}

	if ($fieldParams->get('directory', '/') !== '/')
	{
		$buffer .= sprintf('<img src="images/%s/%s"%s>',
			$fieldParams->get('directory'),
			htmlentities($path, ENT_COMPAT, 'UTF-8', true),
			$class
		);
	}
	else
	{
		$buffer .= sprintf('<img src="images/%s"%s>',
			htmlentities($path, ENT_COMPAT, 'UTF-8', true),
			$class
		);
	}
```

**Comment by PhilETaylor** [comment]
Directory doesn't apply in this, the media custom field, as the media value is already /a/full/path/to/image.jpg

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on a1e0b51adfbdccf01843741baa42f8086ac6e74d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16151">issues.joomla.org/tracker/joomla-cms/16151</a>.</sub>

**Comment by brianteeman** [comment]
There is a lot more than just removing the slash in this PR

**Comment by N6REJ** [comment]
I don't see where the / is added

**Comment by PhilETaylor** [comment]
> I don't see where the / is added back

Because the whole point was to REMOVE it because apparently Joomla now demands PR's that do not close void tags with a closing slash.

**Comment by N6REJ** [comment]
@PhilETaylor ok, then I'm really confuzzled because then we're breaking html?

**Comment by PhilETaylor** [comment]
html doesn’t require the use of a closing slash to close void tags - a quick google will have you thinking you were mad to think it did, and back again. Its optional apparently, but there are those in the joomla world now demanding that PR creators are held to different standards than those contributing a whole new feature (com_fields) apparently.

**Comment by brianteeman** [comment]
@N6REJ no we are not. html5 does not require the /

**Comment by N6REJ** [comment]
@PhilETaylor I think I'll go back to contemplating why the hamster spins the wheel ...  thanks

**Comment by franz-wohlkoenig** [comment]
With and -out PR got:
bildschirmfoto 2017-05-23 um 08 03 38
### System information
3.7.3-dev
Multilanguage Site
macOS Sierra, 10.12.4
Firefox 53 (64-bit)
#### MAMP 4.1.1
- PHP 7.0.15
- MySQLi 5.6.35

**Comment by infograf768** [comment]
@franz-wohlkoenig 
This works for an imagelist custom field.
I get here OK
```
dl class="fields-container">
			<dd class="field-entry ">
			<span class="field-label">imagelist: </span>
<span class="field-value"><img src="/testnew/trunkgitnew/images/joomla_black.png" class="imgclass"></span>
	</dd>
</dl>
```

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on a1e0b51adfbdccf01843741baa42f8086ac6e74d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16151">issues.joomla.org/tracker/joomla-cms/16151</a>.</sub>

**Comment by PhilETaylor** [comment]
@franz-wohlkoenig that is impossible as there is no closing hash in the source code after the PR is applied. What I have seen though, is that Google Chrome Dev Console (and others) likes to fake the html to make it "more valid" and so sometimes adds tags.  Try viewing the source code of the page using "View Source Code" features of your browser instead of using the dev tools panel.

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Comment by franz-wohlkoenig** [comment]
@infograf768 didn't saw its about com_fields.

**Pull request merged**